[RU] Add `Client/Beatmap editor/AiMod` (#14908)
ppy/osu-wiki: wiki/Client/Beatmap_editor/AiMod/ru.md

Несмотря на то, что AiMod обнаруживает базовые ошибки в картах, он не может служить заменой [моддинга](/wiki/Modding) со стороны других игроков и мапперов. AiMod не оценивает качество карты и не обнаруживает сложные проблемы вроде посредственных [паттернов](/wiki/Beatmap/Pattern) или [тайминга](/wiki/Guides/How_to_time_songs). В ходе [маппинга](/wiki/Beatmapping) рекомендуется сначала исправить замечания AiMod, а уже потом искать [моддеров](/wiki/Modding/Modder) на свою карту.

Разработка AiMod отстаёт от стандартов, популярных среди мапперов или закреплённых в [критериях ранкинга](/wiki/Ranking_criteria), а потому некоторые его сообщения [содержат ошибки](#недостатки).

## Вкладки

- **All:** объединяет проблемы со всех вкладок.
- **Compose:** проблемы, связанные с расстановкой объектов на игровом поле. Чтобы увидеть больше предупреждений, отметьте флажок `Check distance snap` (по умолчанию он снят, чтобы не вызывать лаги на картах с большим числом объектов).

> - **Design:** проблемы, связанные с оформлением или интерфейсом игры: фон, сториборд и т.д.

Reviewer: ```suggestion
- **Design:** проблемы, связанные с визуальным оформлением карты: фон, сториборд и т.д.
```
Мне кажется, фоны/сториборды не относятся ни к интерфейсу игры, ни просто к «оформлению».  Может быть так? Но мой вариант мне не нравится на 100%.

Author: не, твой лучше

- **Timing:** проблемы, связанные с расположением нот на временной шкале.

> - **Meta:** проблемы, связанные с настройками сложности.

Reviewer: ```suggestion
- **Meta:** проблемы, связанные с настройками карты в соответствующей вкладке.
```
А если там про романизацию говорится? Технически это в настройках сложности, но не относится к сложности как таковой.

Author: ну так это всё равно *настройки*, и в сложности они тоже фигурируют (таки да, можно поменять романизацию и теги для одной-единственной сложности)

пока не буду применять, можно просто поменять тогда на «с настройками карты»

Author: нет, ладно, лучше уж так

```suggestion
- **Meta:** проблемы, связанные с настройками карты в окне `Song Setup`.
```

- **Mapset:** проблемы, относящиеся ко всей карте в целом.

## Сообщения

::: alert-notice
**Примечание**
Вместо фигурных скобок, например, `{0}`, в сообщениях от AiMod будут подставлены нужные слова или числа.
:::

### Примечания

#### All

| Сообщение | Пояснение | Как исправить | Примечания |
| :-- | :-- | :-- | :-- |
| No problems were found in this map! |  | В карте не найдено серьёзных проблем. Её можно [опубликовать](/wiki/Beatmapping/Beatmap_submission), чтобы другие люди могли её оценить и выявить недочёты. |  |

#### Meta

| Сообщение | Пояснение | Как исправить | Примечания |
| :-- | :-- | :-- | :-- |

> | HP rate for Easy/Normal is suggested to be at least 4. |  | Откройте окно [Song Setup](/wiki/Client/Beatmap_editor/Song_setup), перейдите на вкладку `Difficulty` и поменяйте значение `HP Drain Rate` 4 или выше. | ![](/wiki/shared/mode/mania.png) Только для режима osu!mania |

Reviewer 2: …и поменяйте значение HP Drain Rate **на*** 4 или выше.

> | HP rate for Hard and up is suggested to be at least 7. |  | Откройте окно [Song Setup](/wiki/Client/Beatmap_editor/Song_setup), перейдите на вкладку `Difficulty` и поменяйте значение `HP Drain Rate` 7 или выше. | ![](/wiki/shared/mode/mania.png) Только для режима osu!mania |

Reviewer 2: …и поменяйте значение HP Drain Rate **на*** 7 или выше.

| OD rate is suggested to be at least 5. |  | Откройте окно [Song Setup](/wiki/Client/Beatmap_editor/Song_setup), перейдите на вкладку `Difficulty` и поменяйте значение `Overall Difficulty` на 5 или выше. | ![](/wiki/shared/mode/mania.png) Только для режима osu!mania. Если низкое значение OD выбрано намеренно, это сообщение можно проигнорировать. |
| OD rate for maps with very few sliders is suggested to be at least 7. |  | Откройте окно [Song Setup](/wiki/Client/Beatmap_editor/Song_setup), перейдите на вкладку `Difficulty` и поменяйте значение `Overall Difficulty` на 7 или выше. | ![](/wiki/shared/mode/mania.png) Только для режима osu!mania. Если низкое значение OD выбрано намеренно, это сообщение можно проигнорировать. |
| OD rate for maps with very few sliders is suggested to be at least 8. |  | Откройте окно [Song Setup](/wiki/Client/Beatmap_editor/Song_setup), перейдите на вкладку `Difficulty` и поменяйте значение `Overall Difficulty` на 8 или выше. | ![](/wiki/shared/mode/mania.png) Только для режима osu!mania. Если низкое значение OD выбрано намеренно, это сообщение можно проигнорировать. |

> | The Slider Velocity should be 1.40 or 1.60. | Базовая скорость слайдеров, используемая в карте, должна быть равна 1.40 или 1.60. Иные значения нарушают рекомендации [osu!taiko ranking criteria](/wiki/Ranking_criteria/osu!taiko) по оптимальному расстоянию между нотами. | Перейдите на вкладку Timing и выставите значение `Slider Velocity:` в 1.40 или 1.60. | ![](/wiki/shared/mode/taiko.png) Только для режима osu!taiko |

Reviewer 2: ... и выставите значение Slider Velocity: **на*** 1.40 или 1.60.

Author: поменять *на* ..., **но** выставить *в* ... (или выставить ..., *равное* ...)

#### Design

| Сообщение | Пояснение | Как исправить | Примечания |
| :-- | :-- | :-- | :-- |
| Your beatmap has no background image. |  | Подберите подходящее изображение и используйте его в качестве фона для сложности. | Видео не считается фоновым изображением, поскольку карту можно скачать и без видео. |
| File missing:{0} | В карте отсутствуют некоторые файлы, на которые она ссылается. | Восстановите недостающие файлы или убедитесь, что карта не пытается использовать несуществующие файлы. |  |

#### Timing

| Сообщение | Пояснение | Как исправить | Примечания |
| :-- | :-- | :-- | :-- |
| All timing sections have a volume below 5%. |  | Установите хотя бы в одной тайминг-секции громкость хитсаундов на уровне 5% или выше. |  |

#### Meta

| Сообщение | Пояснение | Как исправить | Примечания |
| :-- | :-- | :-- | :-- |

> | Drain time should be over 30 seconds. | Длина карты от начала до конца без учёта перерывов меньше 30 секунд. | Завершите карту спиннером, если это подходит песне. Если же песня попросту слишком короткая, удлините файл `.mp3`, чтобы дотянуть до отметки в 30 секунд. | AiMod не учитывает спиннеры, которые могут продлевать игровое время за отметку в 30 секунд. |

Reviewer: ```suggestion
| Drain time should be over 30 seconds. | Длина карты от начала до конца без учёта перерывов меньше 30 секунд. | Завершите карту спиннером, если это подходит песне. Если же песня попросту слишком короткая, удлините файл `.mp3`, чтобы дотянуть до отметки в 30 секунд. | AiMod не учитывает спиннеры, но они могут продлевать игровое время за отметку в 30 секунд. |
```
Здесь возражение в виде «но», чтобы явно указать, что спиннер в исключениях аимода.

### Предупреждения

#### Compose

| Сообщение | Пояснение | Как исправить | Примечания |
| :-- | :-- | :-- | :-- |

> | This slider moves in an abnormal way. | Слайдер накладывается сам на себя, возвращаясь точно по тому же пути, по которому шёл (такие слайдеры называют [бураи-слайдерами](/wiki/Beatmapping/Mapping_techniques/Unrankable#burai-sliders)). | Проследите, чтобы слайдер не накладывался сам на себя. |  |

Reviewer 2: ... (такие слайдеры называют [**бурай***-слайдерами](...)). (Если, конечно, слово взято из японского языка)

| Slider has an absurdly large amount of points! |  | Удалите часть контрольных точек слайдера. | Это предупреждение можно проигнорировать, если большое число контрольных точек добавлено специально — например, в сложных [слайдер-артах](http://osu.ppy.sh/community/forums/topics/689531). |

> | This combo is very long. Consider splitting it up. |  | Разбейте длинное комбо на несколько более коротких. Желательно, чтобы в каждом комбо было не больше 15–18 объектов. | Особенно заметно на картах ![](/wiki/shared/mode/catch.png) osu!catch, где фрукты будут скапливаться на тарелке ловца, пока комбо не прервётся, из-за чего может ухудшаться обзор. Это предупреждение можно проигнорировать, если длинное комбо используется намеренно. |

Reviewer: ```suggestion
| This combo is very long. Consider splitting it up. |  | Разбейте длинное комбо на несколько более коротких. Желательно, чтобы в каждом комбо было не больше 15–18 объектов. | Особенно заметно на картах ![](/wiki/shared/mode/catch.png) osu!catch, где фрукты будут скапливаться на тарелке ловца, пока комбо не прервётся, а это может затруднить обзор. Это предупреждение можно проигнорировать, если длинное комбо используется намеренно. |
```

| This map may need an epilepsy warning, as it contains frequently toggled storyboards. | Сториборд содержит быстро мерцающие объекты, которые могут спровоцировать приступ [эпилепсии](https://ru.wikipedia.org/wiki/Эпилепсия) у некоторых игроков. | Откройте окно [Song Setup](/wiki/Client/Beatmap_editor/Song_setup), перейдите на вкладку `Design` и включите опцию `Display epilepsy warning (storyboard has quick strobing)`. |  |
| <!-- Not sure how to test this one -->{0}'s dimensions must be {1}x{1} | Размер указанного элемента не соответствует ожидаемому. | Измените разрешение этого элемента до нужного. |  |
| Your video's dimensions must not exceed 1024x768 for the 4:3 format. |  | Уменьшите или замените фоновое видео. |  |
| Your video's dimensions must not exceed 1280x720 for the 16:9 format. |  | Уменьшите или замените фоновое видео. |  |

#### Timing

| Сообщение | Пояснение | Как исправить | Примечания |
| :-- | :-- | :-- | :-- |
| This beatmap is over 6 minutes long. Consider shortening it if it's not a marathon-style map. |  | Добавьте в карту больше перерывов или обрежьте аудиофайл. | Это предупреждение можно проигнорировать, если карта сделана такой намеренно. |

> | Your beatmap is shorter than 45 seconds. Consider making it longer. |  | Замапайте побольше. | Это предупреждение можно проигнорировать, если карта сделана такой намеренно. |

Reviewer: ```suggestion
| Your beatmap is shorter than 45 seconds. Consider making it longer. |  | Так как ваша карта идёт меньше 45 секунд, замапайте побольше. | Это предупреждение можно проигнорировать, если карта сделана такой намеренно. |
```
Я бы частично перевёл просто, почему именно надо замаппать побольше, контекст какой-то добавить. Не все же знают английский.

Author: ну допустим
(если что, я убрал двойную п из твоего предложения)

| Kiai needs an end time point. |  | Добавьте новую зелёную тайминг-секцию, которая будет выключать киаи. |  |
| A preview point for this map is not set. Consider setting one from the Timing menu. |  | Настройте аудиопревью карты через `Timing` > `Set Current Position as Preview Point`. |  |
| Two timing points exist at the same time! |  | Удалите одну из конфликтующих тайминг-секций. |  |
| {0} out of {1} timing sections have a volume below 5%. |  | Установите громкость хитсаундов в тихих тайминг-секциях на уровне 5% или выше. |  |
| More than 1/3 of the map is Kiai time. Consider reducing this. |  | Сократите использование киаи на карте. | Это предупреждение можно проигнорировать, если киаи расставлен так намеренно. |
| More than 1/2 of the TV Size map is Kiai time. Consider reducing this. |  | Сократите использование киаи на карте. | Это предупреждение можно проигнорировать, если киаи расставлен так намеренно. |
| Kiai isn't snapped! | Начало киаи не привязано к временно́й шкале. | Поставьте начало киаи на нужное деление временно́й шкалы. |  |

> | Kiai's end isn't snapped! | Конец указанного киаи не привязана к временно́й шкале. | Поставьте конец киаи на нужное деление временно́й шкалы. |  |

Reviewer 2: Конец указанного киаи **не привязан*** к временной шкале.

| This mapset needs at least 2 osu!mania diffs. |  | Добавьте в карту вторую сложность. | ![](/wiki/shared/mode/mania.png) Только для режима osu!mania. Это предупреждение основано на правиле, которое давно неактуально. Его можно проигнорировать, если карта соответствует [требованиям к подборке сложностей](/wiki/Ranking_criteria/osu!mania#правила) из критериев ранкинга. |

## Недостатки

- **AiMod не способен обнаружить множество проблем**, которые перечислены в [критериях ранкинга](/wiki/Ranking_criteria), например, объекты с неправильной привязкой к шкале, запаздывающие хитсаунды, фон размером больше 2,5 МБ и так далее.
- **AiMod может предупреждать о том, что проблемой не является** — например, что в карту-[марафон](/wiki/Beatmap/Marathon) надо добавить ещё сложностей, или что [киаи](/wiki/Gameplay/Kiai_time) включён менее чем на 15 секунд.
- **Если включена опция `Check distance snap`, AiMod всё равно не проверяет пропорциональное расстояние между соседними комбо.**
- **Если в папке карты лежат несколько файлов MP3, AiMod будет считать, что они все являются песнями**, хотя MP3 может использоваться и для звуковых эффектов — например, аплодисментов.

> Если вы хотите [ранкнуть](/wiki/Beatmap/Category#ranked) свою карты, то, помимо AiMod, рекомендуется пользоваться и другими программами — например, [Mapset Verifier](https://github.com/Naxesss/MapsetVerifier/releases) от ::{ flag=SE }:: [Naxess](https://osu.ppy.sh/users/8129817). Такие инструменты лучше соответствуют текущим критериям ранкинга и находят гораздо больше проблем, чем AiMod. Тем не менее, не игнорируйте AiMod целиком и полностью, поскольку иногда он всё-таки предупреждает о вещах, запрещённых критериями ранкинга.

Reviewer 2: ...если вы хотите ранкнуть свою **карту*, то** помимо AiMod рекомендуется пользоваться...  

Свою карту* - здесь понятно.   

С запятыми (окончательный вариант): "Если вы хотите ранкнуть свою карту, то помимо AiMod рекомендуется пользоваться и другими программами..."

Author: https://gramota.ru/biblioteka/spravochniki/spravochnik-po-punktuatsii/pomimo-chego

> Если вы хотите [ранкнуть](/wiki/Beatmap/Category#ranked) свою карты, то, помимо AiMod, рекомендуется пользоваться и другими программами — например, [Mapset Verifier](https://github.com/Naxesss/MapsetVerifier/releases) от ::{ flag=SE }:: [Naxess](https://osu.ppy.sh/users/8129817). Такие инструменты лучше соответствуют текущим критериям ранкинга и находят гораздо больше проблем, чем AiMod. Тем не менее, не игнорируйте AiMod целиком и полностью, поскольку иногда он всё-таки предупреждает о вещах, запрещённых критериями ранкинга.

Reviewer: ```suggestion
Если вы хотите [ранкнуть](/wiki/Beatmap/Category#ranked) свою карты, то, помимо AiMod, рекомендуется пользоваться и другими программами — например, [Mapset Verifier](https://github.com/Naxesss/MapsetVerifier/releases) от ::{ flag=SE }:: [Naxess](https://osu.ppy.sh/users/8129817). В них проверка идёт по более актуальным критериям ранкинга, и они находят гораздо больше проблем, чем AiMod. Тем не менее, не игнорируйте AiMod целиком и полностью, поскольку иногда он всё-таки предупреждает о вещах, запрещённых критериями ранкинга.
```

Author: таки да
